tensorflow/core/kernels/data/dataset_test_base.cc

```diff
@@ -0,0 +1,182 @@
+  params.function_handle_cache = new FunctionHandleCache(flr_);
+  iterator_context->reset(new IteratorContext(params));
+  std::unique_ptr<IteratorBase> iterator;
+  TF_CHECK_OK(
+      dataset->MakeIterator(iterator_context->get(), "Iterator", &iterator));
+  return iterator;
+}
+
+Status DatasetOpsTestBase::GetNext(IteratorBase* iterator,
+                                   IteratorContext* iterator_context,
+                                   std::vector<Tensor>* out_tensors,
+                                   bool* end_of_sequence) {
+  return iterator->GetNext(iterator_context, out_tensors, end_of_sequence);
+}
+
+DatasetBase* DatasetOpsTestBase::GetDatasetFromContext(OpKernelContext* context,
+                                                       int output_index) {
+  DatasetBase* dataset;
+  Tensor* output = context->mutable_output(output_index);
+  TF_CHECK_OK(GetDatasetFromVariantTensor(*output, &dataset));
+  dataset->Ref();
+  return dataset;
+}
+
+Status DatasetOpsTestBase::InitThreadPool(int thread_num) {
+  if (thread_num < 1) {
+    return errors::InvalidArgument(
+        "The `thread_num` argument should be but got: ", thread_num);
+  }
+  thread_pool_ = MakeUnique<thread::ThreadPool>(Env::Default(), ThreadOptions(),
+                                                "inter_op", thread_num);
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::InitFunctionLibraryRuntime(
+    const std::vector<FunctionDef>& flib, int cpu_num) {
+  if (cpu_num < 1) {
+    return errors::InvalidArgument(
+        "The `cpu_num` argument should be positive but got: ", cpu_num);
+  }
+  SessionOptions options;
+  auto* device_count = options.config.mutable_device_count();
+  device_count->insert({"CPU", cpu_num});
+  std::vector<std::unique_ptr<Device>> devices;
+  TF_RETURN_IF_ERROR(DeviceFactory::AddDevices(
+      options, "/job:localhost/replica:0/task:0", &devices));
+  device_mgr_ = MakeUnique<DeviceMgr>(std::move(devices));
+
+  FunctionDefLibrary proto;
+  for (const auto& fdef : flib) *(proto.add_function()) = fdef;
+  lib_def_ = MakeUnique<FunctionLibraryDefinition>(OpRegistry::Global(), proto);
+
+  OptimizerOptions opts;
+  pflr_ = MakeUnique<ProcessFunctionLibraryRuntime>(
+      device_mgr_.get(), Env::Default(), TF_GRAPH_DEF_VERSION, lib_def_.get(),
+      opts, thread_pool_.get(), nullptr /* cluster_flr */);
+  flr_ = pflr_->GetFLR("/job:localhost/replica:0/task:0/cpu:0");
+  if (thread_pool_ == nullptr) {
+    runner_ = [](std::function<void()> fn) { fn(); };
+  } else {
+    runner_ = [this](std::function<void()> fn) {
+      thread_pool_->Schedule(std::move(fn));
+    };
+  }
+  return Status::OK();
+}
+
+Status DatasetOpsTestBase::RunOpKernel(OpKernel* op_kernel,
+                                       OpKernelContext* context) {
+  device_->Compute(op_kernel, context);
+  return context->status();
+}
+
+std::unique_ptr<OpKernelContext> DatasetOpsTestBase::CreateOpKernelContext(
```

Comment:
Although this function can never fail, I would still change its signature to return `Status` to make it possible to add something to the function body that could fail.